core/src/View/MDSpan/Kokkos_MDSpan_Layout.hpp

```diff
@@ -0,0 +1,164 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_IMPL_PUBLIC_INCLUDE
+static_assert(false,
+              "Including non-public Kokkos header files is not allowed.");
+#endif
+
+#ifndef KOKKOS_EXPERIMENTAL_MDSPAN_LAYOUT_HPP
+#define KOKKOS_EXPERIMENTAL_MDSPAN_LAYOUT_HPP
+
+#include "Kokkos_MDSpan_Extents.hpp"
+#include <Kokkos_Core_fwd.hpp>
+#include <impl/Kokkos_ViewDataAnalysis.hpp>
+
+namespace Kokkos::Impl {
+template <class Layout>
+struct ArrayLayoutFromLayout;
+
+template <std::size_t padding_value>
+struct ArrayLayoutFromLayout<Experimental::layout_left_padded<padding_value>> {
+  using type = Kokkos::LayoutLeft;
+  static constexpr std::integral_constant<unsigned,
+                                          static_cast<unsigned>(padding_value)>
+      padding = {};
+};
+
+template <std::size_t padding_value>
+struct ArrayLayoutFromLayout<Experimental::layout_right_padded<padding_value>> {
+  using type = Kokkos::LayoutRight;
+  static constexpr std::integral_constant<unsigned,
+                                          static_cast<unsigned>(padding_value)>
+      padding = {};
+};
+
+template <>
+struct ArrayLayoutFromLayout<layout_stride> {
+  using type = Kokkos::LayoutStride;
+};
+
+template <class ArrayLayout>
+struct LayoutFromArrayLayout;
+
+template <>
+struct LayoutFromArrayLayout<Kokkos::LayoutLeft> {
+  using type = Experimental::layout_left_padded<dynamic_extent>;
+};
+
+template <>
+struct LayoutFromArrayLayout<Kokkos::LayoutRight> {
+  using type = Experimental::layout_right_padded<dynamic_extent>;
+};
+
+template <>
+struct LayoutFromArrayLayout<Kokkos::LayoutStride> {
+  using type = layout_stride;
+};
+
+template <class T, class Extents, class Layout>
+struct ViewOffsetFromExtents {
```

Comment:
The name of this trait is misleading.
```suggestion
struct ViewOffsetFromExtentsAndLayout {
```